File: 02_ProgrammingFundamentals/_06_ObjectsAndClasses/_01_Lecture/_03_BigFactorial.java
package _02_ProgrammingFundamentals._06_ObjectsAndClasses._01_Lecture;

import java.math.BigDecimal;
import java.util.Scanner;

public class _03_BigFactorial {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input value
        int factorial = Integer.parseInt(scan.nextLine());

        // Add variable
        BigDecimal result = new BigDecimal(1);

        // Determine the factorial of the input
        for (int i = 1; i <= factorial; i++) {
            result = result.multiply(new BigDecimal(String.valueOf(i)));
        }

        // Print result
        System.out.println(result);

        scan.close();
    }
}

File: 24_C#Fundamentals/_06_ObjectsAndClasses/_01_Lecture/_03_BigFactorial/_03_BigFactorial.cs
using System;
using System.Numerics;

public class _03_BigFactorial
{
    public static void Main(string[] args)
    {
        // Read input value
        int factorial = int.Parse(Console.ReadLine());

        // Add variable
        BigInteger result = BigInteger.One;

        // Determine the factorial of the input
        for (int i = 1; i <= factorial; i++)
        {
            result *= i;
        }

        // Print result
        Console.WriteLine(result);
    }
}
